src/main/java/com/google/api/codegen/viewmodel/OptionalArrayMethodView.java

```diff
@@ -71,6 +73,14 @@ public static Builder newBuilder() {
     return new AutoValue_OptionalArrayMethodView.Builder();
   }
 
+  public boolean isReturnEmpty() {
+    return !hasReturnValue();
+  }
+
+  public boolean hasRequestStreaming() {
```

Comment:
It means either client streaming or bidi streaming. In either case, it's a 'streaming' request.